### ISSUE TITLE
Blank stylesheet after installation

### ISSUE THREAD

**Issue by ukcb** [Nordaaker/convos#335]
I do not know how to describe it ...

etchbox:/home/local# ls -ld convos
drwxr-xr-x 10 www-data www-data 4096 Apr 16 12:54 convos

etchbox:/home/local# cp -v convos/config.json /tmp
„convos/config.json“ -> „/tmp/config.json“

etchbox:/home/local# service convos status
● convos.service - Convos Backend
   Loaded: loaded (/etc/systemd/system/convos.service; static)
   Active: active (running) since So 2017-04-16 13:51:23 CEST; 1h 8min ago
 Main PID: 20678 (perl)
   CGroup: /system.slice/convos.service
           └─20678 perl /home/local/convos/script/convos /home/local/convos/config.json daemon --listen http://127.0.0.1:3001 &>>/var/www/convos/convos.log

Apr 16 13:51:23 etchbox systemd[1]: Starting Convos Backend...
Apr 16 13:51:23 etchbox systemd[1]: Started Convos Backend.
Apr 16 13:51:24 etchbox convos[20678]: [Sun Apr 16 13:51:24 2017] [info] CONVOS_HOME="/var/www/convos" # https://convos.by/doc/config.html#convos_home"
Apr 16 13:51:24 etchbox convos[20678]: [Sun Apr 16 13:51:24 2017] [info] Listening at "http://127.0.0.1:3001"
Apr 16 13:51:24 etchbox convos[20678]: Server available at http://127.0.0.1:3001

etchbox:/home/local# service convos stop

etchbox:/home/local# service convos status
● convos.service - Convos Backend
   Loaded: loaded (/etc/systemd/system/convos.service; static)
   Active: inactive (dead)

Apr 16 15:00:45 etchbox systemd[1]: Stopping Convos Backend...
Apr 16 15:00:45 etchbox systemd[1]: Stopped Convos Backend.

**I now delete my installation!**

etchbox:/home/local# rm -rf convos

<pre>
etchbox:/home/local# curl https://convos.by/install.sh | sh -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1186    0  1186    0     0   5034      0 --:--:-- --:--:-- --:--:--  5046
> /usr/bin/git clone --branch stable https://github.com/Nordaaker/convos.git
Klone nach 'convos'...
remote: Counting objects: 24701, done.
remote: Compressing objects: 100% (444/444), done.
remote: Total 24701 (delta 247), reused 0 (delta 0), pack-reused 24237
Empfange Objekte: 100% (24701/24701), 16.20 MiB | 1.51 MiB/s, Fertig.
Löse Unterschiede auf: 100% (14656/14656), Fertig.
Prüfe Konnektivität... Fertig.
> /usr/bin/perl convos/script/convos install
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org File::HomeDir
--> Working on File::HomeDir
Fetching https://cpan.metacpan.org/authors/id/R/RE/REHSACK/File-HomeDir-1.002.tar.gz ... OK
Configuring File-HomeDir-1.002 ... OK
==> Found dependencies: File::Which
--> Working on File::Which
Fetching https://cpan.metacpan.org/authors/id/P/PL/PLICEASE/File-Which-1.21.tar.gz ... OK
Configuring File-Which-1.21 ... OK
Building File-Which-1.21 ... OK
Successfully installed File-Which-1.21
Building File-HomeDir-1.002 ... OK
Successfully installed File-HomeDir-1.002
2 distributions installed
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org File::ReadBackwards
--> Working on File::ReadBackwards
Fetching https://cpan.metacpan.org/authors/id/U/UR/URI/File-ReadBackwards-1.05.tar.gz ... OK
Configuring File-ReadBackwards-1.05 ... OK
Building File-ReadBackwards-1.05 ... OK
Successfully installed File-ReadBackwards-1.05
1 distribution installed
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org JSON::Validator
--> Working on JSON::Validator
Fetching https://cpan.metacpan.org/authors/id/J/JH/JHTHORSEN/JSON-Validator-0.97.tar.gz ... OK
Configuring JSON-Validator-0.97 ... OK
==> Found dependencies: Mojolicious, Test::More
--> Working on Mojolicious
Fetching https://cpan.metacpan.org/authors/id/S/SR/SRI/Mojolicious-7.30.tar.gz ... OK
Configuring Mojolicious-7.30 ... OK
==> Found dependencies: IO::Socket::IP
--> Working on IO::Socket::IP
Fetching https://cpan.metacpan.org/authors/id/P/PE/PEVANS/IO-Socket-IP-0.39.tar.gz ... OK
Configuring IO-Socket-IP-0.39 ... OK
Building IO-Socket-IP-0.39 ... OK
Successfully installed IO-Socket-IP-0.39 (upgraded from 0.32)
Building Mojolicious-7.30 ... OK
Successfully installed Mojolicious-7.30 (upgraded from 5.54)
--> Working on Test::More
Fetching https://cpan.metacpan.org/authors/id/E/EX/EXODIST/Test-Simple-1.302083.tar.gz ... OK
Configuring Test-Simple-1.302083 ... OK
Building Test-Simple-1.302083 ... OK
Successfully installed Test-Simple-1.302083 (upgraded from 1.001008)
Building JSON-Validator-0.97 ... OK
Successfully installed JSON-Validator-0.97
4 distributions installed
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org LinkEmbedder
--> Working on LinkEmbedder
Fetching https://cpan.metacpan.org/authors/id/J/JH/JHTHORSEN/LinkEmbedder-0.04.tar.gz ... OK
Configuring LinkEmbedder-0.04 ... OK
==> Found dependencies: Test::Deep
--> Working on Test::Deep
Fetching https://cpan.metacpan.org/authors/id/R/RJ/RJBS/Test-Deep-1.126.tar.gz ... OK
Configuring Test-Deep-1.126 ... OK
Building Test-Deep-1.126 ... OK
Successfully installed Test-Deep-1.126 (upgraded from 0.113)
Building LinkEmbedder-0.04 ... OK
Successfully installed LinkEmbedder-0.04
2 distributions installed
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org Mojo::IOLoop::ForkCall
--> Working on Mojo::IOLoop::ForkCall
Fetching https://cpan.metacpan.org/authors/id/J/JB/JBERGER/Mojo-IOLoop-ForkCall-0.17.tar.gz ... OK
Configuring Mojo-IOLoop-ForkCall-0.17 ... OK
==> Found dependencies: IO::Pipely
--> Working on IO::Pipely
Fetching https://cpan.metacpan.org/authors/id/R/RC/RCAPUTO/IO-Pipely-0.005.tar.gz ... OK
Configuring IO-Pipely-0.005 ... OK
Building IO-Pipely-0.005 ... OK
Successfully installed IO-Pipely-0.005
Building Mojo-IOLoop-ForkCall-0.17 ... OK
Successfully installed Mojo-IOLoop-ForkCall-0.17
2 distributions installed
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org Mojo::IRC
--> Working on Mojo::IRC
Fetching https://cpan.metacpan.org/authors/id/J/JH/JHTHORSEN/Mojo-IRC-0.44.tar.gz ... OK
Configuring Mojo-IRC-0.44 ... OK
==> Found dependencies: Parse::IRC, IRC::Utils
--> Working on Parse::IRC
Fetching https://cpan.metacpan.org/authors/id/B/BI/BINGOS/Parse-IRC-1.22.tar.gz ... OK
Configuring Parse-IRC-1.22 ... OK
Building Parse-IRC-1.22 ... OK
Successfully installed Parse-IRC-1.22
--> Working on IRC::Utils
Fetching https://cpan.metacpan.org/authors/id/H/HI/HINRIK/IRC-Utils-0.12.tar.gz ... OK
Configuring IRC-Utils-0.12 ... OK
Building IRC-Utils-0.12 ... OK
Successfully installed IRC-Utils-0.12
Building Mojo-IRC-0.44 ... OK
Successfully installed Mojo-IRC-0.44
3 distributions installed
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org Mojolicious
Mojolicious is up to date. (7.30)
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org Mojolicious::Plugin::AssetPack
--> Working on Mojolicious::Plugin::AssetPack
Fetching https://cpan.metacpan.org/authors/id/J/JH/JHTHORSEN/Mojolicious-Plugin-AssetPack-1.41.tar.gz ... OK
Configuring Mojolicious-Plugin-AssetPack-1.41 ... OK
==> Found dependencies: IPC::Run3
--> Working on IPC::Run3
Fetching https://cpan.metacpan.org/authors/id/R/RJ/RJBS/IPC-Run3-0.048.tar.gz ... OK
Configuring IPC-Run3-0.048 ... OK
Building IPC-Run3-0.048 ... OK
Successfully installed IPC-Run3-0.048
Building Mojolicious-Plugin-AssetPack-1.41 ... OK
Successfully installed Mojolicious-Plugin-AssetPack-1.41
2 distributions installed
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org Mojolicious::Plugin::OpenAPI
--> Working on Mojolicious::Plugin::OpenAPI
Fetching https://cpan.metacpan.org/authors/id/J/JH/JHTHORSEN/Mojolicious-Plugin-OpenAPI-1.13.tar.gz ... OK
Configuring Mojolicious-Plugin-OpenAPI-1.13 ... OK
Building Mojolicious-Plugin-OpenAPI-1.13 ... OK
Successfully installed Mojolicious-Plugin-OpenAPI-1.13
1 distribution installed
>>> /usr/bin/perl /home/local/convos/script/cpanm -n -l /home/local/convos/script/../local -M https://cpan.metacpan.org Parse::IRC
Parse::IRC is up to date. (1.22)

-------------------------------
All dependencies are installed!

You can now run "convos/script/convos daemon --listen http://*:8000" to start Convos.
</pre>

etchbox:/home/local# chown -R www-data:www-data convos

etchbox:/home/local# ls -l convos/assets/cache/main-*.css
-rw-r--r-- 1 www-data www-data 131358 Apr 16 15:05 convos/assets/cache/main-ad939c5f7d.css
**-rw-r--r-- 1 www-data www-data  97911 Apr 16 15:05 convos/assets/cache/main-ad939c5f7d.min.css**

etchbox:/home/local# cp -v convos/assets/cache/main-*.css /tmp
„convos/assets/cache/main-ad939c5f7d.css“ -> „/tmp/main-ad939c5f7d.css“
„convos/assets/cache/main-ad939c5f7d.min.css“ -> „/tmp/main-ad939c5f7d.min.css“

etchbox:/home/local# cp -v /tmp/config.json convos/
„/tmp/config.json“ -> „convos/config.json“

etchbox:/home/local# service convos start
etchbox:/home/local# service convos status
● convos.service - Convos Backend
   Loaded: loaded (/etc/systemd/system/convos.service; static)
   Active: active (running) since So 2017-04-16 15:11:38 CEST; 21s ago
 Main PID: 19894 (perl)
   CGroup: /system.slice/convos.service
           └─19894 perl /home/local/convos/script/convos /home/local/convos/config.json daemon --listen http://127.0.0.1:3001 &>>/var/www/convos/convos.log

Apr 16 15:11:38 etchbox systemd[1]: Started Convos Backend.
Apr 16 15:11:38 etchbox convos[19894]: [Sun Apr 16 15:11:38 2017] [info] CONVOS_HOME="/var/www/convos" # https://convos.by/doc/config.html#convos_home"
Apr 16 15:11:40 etchbox convos[19894]: **NoMethodError: undefined method `specificity' for [:not(.optgroup)]:Array**
Apr 16 15:11:40 etchbox convos[19894]: [Sun Apr 16 15:11:40 2017] [info] Listening at "http://127.0.0.1:3001"

etchbox:/home/local# ls -l convos/assets/cache/main-*.css
-rw-r--r-- 1 www-data www-data 131358 Apr 16 15:05 convos/assets/cache/main-ad939c5f7d.css
**-rw-r--r-- 1 www-data www-data      0 Apr 16 15:11 convos/assets/cache/main-ad939c5f7d.min.css**

**At this point, the stylesheet is empty!**

![convos](https://cloud.githubusercontent.com/assets/22983914/25071751/8b87b0f8-22bf-11e7-87f9-cd8d59043265.jpg)

**I use the original style and restarte convos. After that, the style in the browser is ok.**

etchbox:/home/local# cp -v /tmp/main-ad939c5f7d.min.css convos/assets/cache
„/tmp/main-ad939c5f7d.min.css“ -> „convos/assets/cache/main-ad939c5f7d.min.css“

etchbox:/home/local# service convos restart
etchbox:/home/local# service convos status
● convos.service - Convos Backend
   Loaded: loaded (/etc/systemd/system/convos.service; static)
   Active: active (running) since So 2017-04-16 15:18:21 CEST; 43s ago
 Main PID: 22204 (perl)
   CGroup: /system.slice/convos.service
           └─22204 perl /home/local/convos/script/convos /home/local/convos/config.json daemon --listen http://127.0.0.1:3001 &>>/var/www/convos/convos.log

Apr 16 15:18:21 etchbox systemd[1]: Starting Convos Backend...
Apr 16 15:18:21 etchbox systemd[1]: Started Convos Backend.
Apr 16 15:18:22 etchbox convos[22204]: [Sun Apr 16 15:18:22 2017] [info] CONVOS_HOME="/var/www/convos" # https://convos.by/doc/config.html#convos_home"
Apr 16 15:18:22 etchbox convos[22204]: [Sun Apr 16 15:18:22 2017] [info] Listening at "http://127.0.0.1:3001"

etchbox:/home/local# ls -l convos/assets/cache/main-*.css
-rw-r--r-- 1 www-data www-data 131358 Apr 16 15:05 convos/assets/cache/main-ad939c5f7d.css
-rw-r--r-- 1 www-data www-data  97911 Apr 16 15:17 convos/assets/cache/main-ad939c5f7d.min.css

My system is debian/jessie (stable).

**Comment by ukcb** [comment]
**I have updated my ruby-sass** from version 3.4.6-2 to 3.4.23-1~bpo8+1 (debian/backports).
The Stylesheet now works without errors. It is not created minimally, but this seems to be
a problem of the ruby-sass.

etchbox:/home/local# ls -l convos/assets/cache/main-*.css
-rw-r--r-- 1 www-data www-data 131358 Apr 17 09:56 convos/assets/cache/main-ad939c5f7d.css
-rw-r--r-- 1 www-data www-data 131248 Apr 17 09:58 convos/assets/cache/main-ad939c5f7d.min.css

Thank you for your understanding. :-)

**Comment by jhthorsen** [comment]
This is a bug. You should not require sass to run the installer. There must be something wrong in the "make a new release" process.

**Comment by jhthorsen** [comment]
Version 0.99_32 (just made a new release) should not required a sass compiler

**Comment by ukcb** [comment]
I have the version 0.99_32 just re-installed completely and see at the first start a process in the background:
/usr/bin/ruby /usr/bin/sass -s -I /home/local/convos/script/../assets/sass -I /home/local/convos/script/../assets --scss

Why ruby, I think you used perl? ;-)

**Comment by jhthorsen** [comment]
You can use perl all the way, if you install CSS::Sass: `cpanm CSS::Sass`, but it should not be required if you don't have any custom stylesheets. Do you have either `custom.scss` or `variables.scss` in your `$CONVOS_HOME` directory?

**Comment by jhthorsen** [comment]
It's really hard to follow what you're doing in the issue above.

**Comment by jhthorsen** [comment]
Hm... I see now that the production assets where not built! I don't get why this happens. https://github.com/Nordaaker/convos/blob/master/t/production-resources.t is supposed to make sure this is always done.

Anyways: 92363a8 is now pushed to "stable" and "master", so it should be OK now.

**Comment by jhthorsen** [comment]
My best theory is that there was a race condition when I ran the test suite in parallell, using `prove -l -j5`.

**Comment by ukcb** [comment]
I have checked ot the master https://github.com/Nordaaker/convos/commit/92363a8c and **it works**! I haven't either custom.scss or variables.scss in my $CONVOS_HOME. Many thanks!

**Comment by jhthorsen** [comment]
Thank you for your patience 😄
